extractor/filesystem/language/python/uvlock/uvlock.go

```diff
@@ -0,0 +1,147 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package uvlock extracts uv.lock files.
+package uvlock
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/BurntSushi/toml"
+	"github.com/google/osv-scalibr/extractor"
+	"github.com/google/osv-scalibr/extractor/filesystem"
+	"github.com/google/osv-scalibr/extractor/filesystem/language/python/internal/pypipurl"
+	"github.com/google/osv-scalibr/extractor/filesystem/osv"
+	"github.com/google/osv-scalibr/plugin"
+	"github.com/google/osv-scalibr/purl"
+)
+
+type uvLockPackageSource struct {
+	Virtual  string `toml:"virtual"`
+	Git      string `toml:"git"`
+	Registry string `toml:"registry"`
+}
+
+type uvLockPackage struct {
+	Name    string              `toml:"name"`
+	Version string              `toml:"version"`
+	Source  uvLockPackageSource `toml:"source"`
+
+	// uv stores "groups" as a table under "package" after all the packages, which due
+	// to how TOML works means it ends up being a property on the last package, even
+	// through in this context it's a global property rather than being per-package
+	Groups map[string][]uvOptionalDependency `toml:"optional-dependencies"`
+}
+
+type uvOptionalDependency struct {
+	Name string `toml:"name"`
+}
+type uvLockFile struct {
+	Version  int                               `toml:"version"`
+	Packages []uvLockPackage                   `toml:"package"`
+	Groups   map[string][]uvOptionalDependency `toml:"package.optional-dependencies"`
```

Comment:
This also seems to be unused (AFAICS we read uvLockPackage.Groups not uvLockFile.Groups)

Reply:
ah yes good catch - that's left over from my first attempt at extracting the info